src/backend/joanie/tests/core/models/order/test_factory.py

```diff
@@ -0,0 +1,172 @@
+"""Test suite for the OrderGeneratorFactory."""
+
+from django.test import TestCase, override_settings
+
+from joanie.core.enums import (
+    ORDER_STATE_ASSIGNED,
+    ORDER_STATE_CANCELED,
+    ORDER_STATE_COMPLETED,
+    ORDER_STATE_DRAFT,
+    ORDER_STATE_FAILED_PAYMENT,
+    ORDER_STATE_NO_PAYMENT,
+    ORDER_STATE_PENDING,
+    ORDER_STATE_PENDING_PAYMENT,
+    ORDER_STATE_TO_SAVE_PAYMENT_METHOD,
+    ORDER_STATE_TO_SIGN,
+    PAYMENT_STATE_PAID,
+    PAYMENT_STATE_PENDING,
+    PAYMENT_STATE_REFUSED,
+)
+from joanie.core.factories import OrderGeneratorFactory
+
+
+@override_settings(
+    JOANIE_PAYMENT_SCHEDULE_LIMITS={
+        5: (30, 70),
+        10: (30, 45, 45),
+        100: (20, 30, 30, 20),
+    },
+    DEFAULT_CURRENCY="EUR",
+)
+class TestOrderGeneratorFactory(TestCase):
+    """Test suite for the OrderGeneratorFactory."""
+
+    # pylint: disable=too-many-arguments
+    # ruff: noqa: PLR0913
+    def check_order(
+        self,
+        state,
+        has_organization,
+        has_unsigned_contract,
+        is_free,
+        has_payment_method,
+    ):
+        """Check the properties of an order based on the provided parameters."""
+        order = OrderGeneratorFactory(state=state, product__price=100)
+        if has_organization:
+            self.assertIsNotNone(order.organization)
+        else:
+            self.assertIsNone(order.organization)
```

Comment:
I will not deal with specific case in this function. Everything in a `if/else` condition should be moved in the test himself. WDYT ?